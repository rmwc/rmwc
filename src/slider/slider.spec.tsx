import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Slider } from './';

jest.spyOn(console, 'warn');

describe('Slider', () => {
  it('renders uncontrolled', () => {
    const { asFragment } = render(<Slider />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('warns when not displayMarkers and discrete', () => {
    render(<Slider displayMarkers />);
    expect(console.warn).toHaveBeenCalled();
  });

  it('renders with all props', () => {
    const { asFragment } = render(
      <Slider
        value={50}
        discrete
        displayMarkers
        min={0}
        max={100}
        step={10}
        disabled
        onChange={() => {}}
        onInput={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles onChange', async () => {
    let value = 0;
    render(<Slider onChange={() => value++} />);
    userEvent.click(screen.getByRole('slider'));
    await waitFor(() => expect(value).toEqual(1));
  });

  it('handles onInput', async () => {
    let value = 0;
    render(<Slider onInput={() => value++} />);
    userEvent.click(screen.getByRole('slider'));
    await waitFor(() => expect(value).toEqual(1));
  });

  it('handles min > 100', () => {
    render(<Slider min={101} max={200} />);
  });

  it('handles out of bounds', () => {
    render(<Slider value={0} min={1} max={2} />);
    render(<Slider value={3} min={1} max={2} />);
  });

  it('can have custom classnames', () => {
    const { container } = render(<Slider className={'my-custom-classname'} />);
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });
});
