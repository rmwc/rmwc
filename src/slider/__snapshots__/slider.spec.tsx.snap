// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders uncontrolled 1`] = `
<DocumentFragment>
  <div
    aria-label="Select Value"
    class="mdc-slider"
    role="slider"
    tabindex="0"
  >
    <div
      class="mdc-slider__track-container"
    >
      <div
        class="mdc-slider__track"
      />
    </div>
    <div
      class="mdc-slider__thumb-container"
    >
      <svg
        class="mdc-slider__thumb"
        height="21"
        width="21"
      >
        <circle
          cx="10.5"
          cy="10.5"
          r="7.875"
        />
      </svg>
      <div
        class="mdc-slider__focus-ring"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Slider renders with all props 1`] = `
<DocumentFragment>
  <div
    aria-disabled="true"
    aria-label="Select Value"
    aria-valuemax="100"
    aria-valuenow="50"
    class="mdc-slider mdc-slider--discrete mdc-slider--display-markers mdc-slider--disabled"
    data-step="10"
    role="slider"
    tabindex="0"
  >
    <div
      class="mdc-slider__track-container"
    >
      <div
        class="mdc-slider__track"
      />
      <div
        class="mdc-slider__track-marker-container"
      />
    </div>
    <div
      class="mdc-slider__thumb-container"
    >
      <div
        class="mdc-slider__pin"
      >
        <span
          class="mdc-slider__pin-value-marker"
        />
      </div>
      <svg
        class="mdc-slider__thumb"
        height="21"
        width="21"
      >
        <circle
          cx="10.5"
          cy="10.5"
          r="7.875"
        />
      </svg>
      <div
        class="mdc-slider__focus-ring"
      />
    </div>
  </div>
</DocumentFragment>
`;
